### ISSUE TITLE
failed assertion

### ISSUE THREAD

**Issue by xBeret** [HerculesWS/Hercules#1909]
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Issue Prelude

[//]: # (Please complete these mandatory steps and check the following boxes by putting an `x` inside the brackets _before_ filing your issue)

-  [x] I have not modified the source prior to reproducing this issue.
-  [x] I am using the latest version of Hercules.
-  [x] I am aware that this report will be closed or deleted if it becomes obvious that I am stating the false.

### Description

When using a skills or attacking some monster occurs failed assertion

![Image of Yaktocat](http://i63.tinypic.com/2lag4f6.png)

[//]: # (Description of the problem or issue at length.)
[//]: # (Please specify any battle configuration related to the components of this issue that have been changed from the default values. This will allow quicker determination of the cause of the problem.)

**Branch(es):**
- [x] master
- [ ] other

**Hercules rev. hash/commit:** 

Git revision src: c1c8d6b7424b6673bd444b0905171c06602feaa4

[//]: # (Copy the first 3 lines of the login-server, char-server or map-server startup.)
[//]: # ( [Info]: Hercules 64-bit for Mac OS X )
[//]: # ( [Info]: Git revision src: 'a5918b329ca0826b04dca32ede783586403f58db' )
[//]: # ( [Info]: Git revision scripts: 'a5918b329ca0826b04dca32ede783586403f58db' )

**Comment by dastgir** [comment]
@4144 @MishimaHaruna

**Comment by MishimaHaruna** [comment]
Confirmed. Adding assertion reports:

```
[Error]: --- failed assertion --------------------------------------------
[Error]: skill.c:179: 'idx != 0' in function `skill_get_ele'
[Error]: 0   map-server                          0x0000000109e5f346 assert_report + 150
[Error]: 1   map-server                          0x0000000109dc5525 skill_get_ele + 85
[Error]: 2   map-server                          0x0000000109e1c0d5 status_check_skilluse + 2069
[Error]: 3   map-server                          0x0000000109e4d39d unit_attack + 205
[Error]: 4   map-server                          0x0000000109c9eadc clif_parse + 396
[Error]: 5   map-server                          0x0000000109e6e03a do_sockets + 618
[Error]: 6   map-server                          0x0000000109e76b4a main + 442
[Error]: 7   libdyld.dylib                       0x00007fff531ff145 start + 1
[Error]: --- end failed assertion ----------------------------------------
[Error]: --- failed assertion --------------------------------------------
[Error]: skill.c:179: 'idx != 0' in function `skill_get_ele'
[Error]: 0   map-server                          0x0000000109e5f346 assert_report + 150
[Error]: 1   map-server                          0x0000000109dc5525 skill_get_ele + 85
[Error]: 2   map-server                          0x0000000109e1c0d5 status_check_skilluse + 2069
[Error]: 3   map-server                          0x0000000109e4dacc unit_attack_timer_sub + 348
[Error]: 4   map-server                          0x0000000109e4e255 unit_attack_timer + 69
[Error]: 5   map-server                          0x0000000109e4d5a3 unit_attack + 723
[Error]: 6   map-server                          0x0000000109c9eadc clif_parse + 396
[Error]: 7   map-server                          0x0000000109e6e03a do_sockets + 618
[Error]: 8   map-server                          0x0000000109e76b4a main + 442
[Error]: 9   libdyld.dylib                       0x00007fff531ff145 start + 1
[Error]: --- end failed assertion ----------------------------------------
[Error]: --- failed assertion --------------------------------------------
[Error]: skill.c:191: 'idx != 0' in function `skill_get_nk'
[Error]: 0   map-server                          0x0000000109e5f346 assert_report + 150
[Error]: 1   map-server                          0x0000000109dc564c skill_get_nk + 76
[Error]: 2   map-server                          0x0000000109c76b5d battle_calc_weapon_attack + 11597
[Error]: 3   map-server                          0x0000000109c79737 battle_calc_attack + 263
[Error]: 4   map-server                          0x0000000109c7fddc battle_weapon_attack + 2700
[Error]: 5   map-server                          0x0000000109e4e0d4 unit_attack_timer_sub + 1892
[Error]: 6   map-server                          0x0000000109e4e255 unit_attack_timer + 69
[Error]: 7   map-server                          0x0000000109e4d5a3 unit_attack + 723
[Error]: 8   map-server                          0x0000000109c9eadc clif_parse + 396
[Error]: 9   map-server                          0x0000000109e6e03a do_sockets + 618
[Error]: --- end failed assertion ----------------------------------------
```

**Comment by Nurkz** [comment]
@MishimaHaruna also check out skill->get_splash, if no splash info in skill_db => using skill in party cause stuck
![image](https://user-images.githubusercontent.com/15338159/33189051-cb7ccc7a-d0a7-11e7-9385-8ba783d2e93d.png)

as a temporary suggestion, we will wait for official fix

**Comment by cahya1992** [comment]
and on char server i have some issue too
![char](https://user-images.githubusercontent.com/8091082/33195466-620c9884-d10a-11e7-8199-8dff3c8db068.png)

**Comment by 4144** [comment]
@cahya1992 this can be because one of plugins. Try run witout it.

**Comment by MishimaHaruna** [comment]
Published a v2017.11.19-1 hotfix release that includes #1910 (currently available on the stable branch)
